File: src/main/java/com/arcao/feedback/collector/AppInfoCollector.java
package com.arcao.feedback.collector;

import android.content.Context;
import android.content.pm.PackageInfo;
import org.acra.util.PackageManagerWrapper;

public class AppInfoCollector extends Collector {
	private Context context;

	public AppInfoCollector(Context context) {
		this.context = context.getApplicationContext();
	}

	@Override
	public String getName() {
		return "APP INFO";
	}

	@Override
	protected String collect() {
		final StringBuilder sb = new StringBuilder();

		final PackageManagerWrapper pm = new PackageManagerWrapper(context);

		final PackageInfo pi = pm.getPackageInfo();
		if (pi != null) {
			sb.append("APP_PACKAGE=").append(pi.packageName).append("\n");
			sb.append("APP_VERSION_CODE=").append(pi.versionCode).append("\n");
			sb.append("APP_VERSION_NAME=").append(pi.versionName).append("\n");
		}

		return sb.toString();
	}
}

File: app/src/main/java/com/arcao/feedback/collector/AppInfoCollector.kt
package com.arcao.feedback.collector

import com.arcao.geocaching4locus.App

class AppInfoCollector(private val app: App) : Collector() {
    override val name: String
        get() = "APP INFO"

    override suspend fun collect(): String {
        val sb = StringBuilder()

        sb.append("APP_PACKAGE=").append(app.packageName).append("\n")
        sb.append("APP_VERSION_CODE=").append(app.versionCode).append("\n")
        sb.append("APP_VERSION_NAME=").append(app.version).append("\n")

        return sb.toString()
    }
}
